vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1877 @@
+            VTR_ASSERT(sw_id > 1);
+            int name, wire0, wire1;
+            bool forward;
+            std::tie(name, wire0, wire1, forward) = metadata;
+            t_location loc = tile_loc_bimap_[tile_id];
+            auto virtual_key1 = virtual_redirect_[std::make_tuple(node1, loc)];
+            auto virtual_key2 = virtual_redirect_[std::make_tuple(node2, loc)];
+
+            auto key1 = std::make_tuple(std::get<0>(virtual_key1),
+                                        std::get<1>(virtual_key1),
+                                        virtual_beg_to_real_[virtual_key1]);
+            auto key2 = std::make_tuple(std::get<0>(virtual_key2),
+                                        std::get<1>(virtual_key2),
+                                        virtual_beg_to_real_[virtual_key2]);
+
+            int src, sink;
+            src = loc_type_idx_to_rr_idx_[key1];
+            sink = loc_type_idx_to_rr_idx_[key2];
+            device_ctx_.rr_graph_builder.emplace_back_edge(RRNodeId(src), RRNodeId(sink), sw_id);
+
+            // TODO: find a more efficient way to pack PIP information to build the physical netlist out
+            //       of a routed design
+            char metadata_[100];
+            char* temp = int_to_string(metadata_, name);
+            *temp++ = '.';
+            temp = int_to_string(temp, wire0);
+            *temp++ = ',';
+            temp = int_to_string(temp, wire1);
+            *temp++ = '.';
+            temp = int_to_string(temp, forward ? 1 : 0);
+            *temp++ = 0;
+
+            vpr::add_rr_edge_metadata(src, sink, sw_id, vtr::string_view("FPGAInterchange"), vtr::string_view(metadata_));
+        }
+    }
+
+    void pack_rr_edges() {
+        pack_tiles_edges();
+        pack_chans_edges();
+        pack_pips();
+        device_ctx_.rr_graph_builder.mark_edges_as_rr_switch_ids();
+        device_ctx_.rr_graph_builder.partition_edges();
+    }
+
+    /** @brief Fills the device RR graph.
+     */
+    void pack_to_rr_graph() {
+        pack_rr_nodes();
+        pack_rr_edges();
+    }
+
+    /** @brief Finilizes the RR graph generation.
+     *
+     *      - Builds the node lookup
+     *      - Inits fan_ins
+     *      - Loads indexed data
+     */
+    void finish_rr_generation() {
+        for (t_rr_type rr_type : RR_TYPES)
+            if (rr_type == CHANX)
+                device_ctx_.rr_graph_builder.node_lookup().resize_nodes(grid_.height(), grid_.width(), rr_type, NUM_SIDES);
+            else
+                device_ctx_.rr_graph_builder.node_lookup().resize_nodes(grid_.width(), grid_.height(), rr_type, NUM_SIDES);
+
+        for (size_t node = 0; node < rr_nodes_.size(); node++) {
+            auto rr_node = rr_nodes_[node];
+            device_ctx_.rr_graph_builder.add_node_to_all_locs(rr_node.id());
+        }
+
+        device_ctx_.rr_graph_builder.init_fan_in();
+
+        // Creates a temporary copy to convert from vtr::vector to sitd::vector.
```

Comment:
typo: 
std::vector
Should also file an issue to track changing alloc_and_load_rr_indexed_data to a vtr::vector